kernel/src/net/socket/netlink/addr/mod.rs

```diff
@@ -43,6 +43,15 @@ impl NetlinkSocketAddr {
     pub const fn groups(&self) -> GroupIdSet {
         self.groups
     }
+
+    /// Adds specified groups and returns a new address.
+    pub fn add_groups(&self, mut groups: GroupIdSet) -> NetlinkSocketAddr {
```

Comment:
```rust
    #[must_use]
    pub fn add_groups(&self, mut groups: GroupIdSet) -> NetlinkSocketAddr {
```

This seems to be the only `add_groups` that takes `&self` (instead of `&mut self`) and returns the new address. It's not ideal, but I can't think of a better name.

Reply:
I have change the method to accept `&mut self` instead of `&self` to keep consistent. The caller can called `addr.clone().add_groups()` instead.